config/routes.rb

```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
+  get 'users/posts'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
+    get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+    get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+    get '/users', to: 'users#index', as: 'users'
+    get '/users/:id', to: 'users#show', as: 'user'
```

Comment:
- [ ] **[OPTIONAL]** This routes can be defined in a more brief way.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

Reply:
Thank you so much for your suggestion